src/main/java/au/gov/digitalhealth/terminology/amtflatfile/AmtCache.java

```diff
@@ -82,36 +126,43 @@ private void processAmtFiles(FileSystem amtZip) throws IOException {
         readFile(visitor.getLanguageRefsetFile(), s -> handleLanguageRefsetRow(s), true, "\t");
         readFile(visitor.getDescriptionFile(), s -> handleDescriptionRow(s), true, "\t");
         readFile(visitor.getArtgIdRefsetFile(), s -> handleArtgIdRefsetRow(s), true, "\t");
-        readFile(visitor.getMedicinalProductRefsetFile(), s -> handleMedicinalProductRefsetRow(s), true, "\t");
+        int refsetFileCount = 0;
+        for (Path amtRefsetFile : visitor.getAMTRefsetFiles()) {
+            readFile(amtRefsetFile, s -> handleAMTRefsetRow(s), true, "\t");
+            refsetFileCount++;
+        }
+        if (refsetFileCount > 1) {
+            this.amtV3 = true;
+        }
         for (Path historicalFile : visitor.getHistoricalAssociationRefsetFiles()) {
             readFile(historicalFile, s -> handleHistoricalAssociationRefsetRow(s), true, "\t");
         }
 
+        if (this.isAmtV3()) {
```

Comment:
Is it worth trying to refactor this code to only work with V4 or only V3? I.e. different versions of the flat file generator and we can cut down some code? Not sure how much it would change, maybe not with the effort?